### ISSUE TITLE
Batch size

### ISSUE THREAD

**Issue by nwinch** [Nicktho/batch-promises#3]
Initially I thought that the batch size was the total number of batches to split the array into, instead of an explicit batch size.  Perhaps this could be an additional feature/option?

For example:
```
batchPromises(2, [1, 2, 3, 4, 5, 6])
- batch 1 = 1, 2, 3
- batch 2 = 4, 5, 6
```

So with an array of 100 promises and a batch size of 2, it would batch into two chunks of 50.

Odd numbers would have to be accounted for, perhaps that could always be the last remaining batch eg.
```
batchPromises(2, [1, 2, 3, 4, 5])
- batch 1 = 1, 2
- batch 2 = 3, 4, 5
```

**Comment by Nicktho** [comment]
I thought about this initially, but came to the conclusion that you're probably batching because you know there is some limit to whatever you are doing asynchronously. Changing this to "x amount of batches" means that the size of each batch is unknown, leading to the similar problems I was facing with an unknown item list size.

**Comment by nwinch** [comment]
Yeah that makes sense. Thought I'd run it by you.

**Comment by nwinch** [comment]
Happy for this to be closed.

**Comment by Venryx** [comment]
Interesting. [Async-pool](https://github.com/rxaviers/async-pool) takes an alternate approach, where the number you pass is the "pool size" (ie. the max number of concurrent processings), whereas this one apparently is instead "the number of items for each processor to process"?

I find the "pool size" configuration easier/more-useful for my needs, but I'm sure there are ones where the "number of items for each processor to process" config is easier/more-useful.

I can't think of examples of the latter of the top of my head, though. Do you have examples to help illustrate it? Perhaps worth adding to the readme for a comparison between the two libraries.

**Comment by Nicktho** [comment]
@Venryx When I initially wrote this library, I wasn't able to find a library to handle resolving a maximum amount of concurrent promises at a time, before moving on to further promises in a list of promises. In this way, both this library and https://github.com/rxaviers/async-pool achieve the same design goal: limit the amount of promises "resolving" at any one time for a given list of promises to X.

https://github.com/rxaviers/async-pool extends this design goal to something like: limit and maintain the amount of promises "resolving" at any one time for a given list of promises to X.

For most use-cases I built this library for (eg, api limits on concurrent connections), https://github.com/rxaviers/async-pool would be the better pick. I would of used that library if I had found it at the time of writing this one.

As for edge cases where this library _could_ be useful as an alternative, off the top of my head I can only think of a combination of concurrency limits and order importance, although the same can be achieved with https://github.com/rxaviers/async-pool and a reduce. Something like: 

```js
batchPromises(3, items, callApiWithItem).then(() => {
  // `callApiWithItem` is guaranteed to be resolved with `items[0]`, `items[1]`, `items[2]` before `items[3]` etc. 
})
```

As for a real-world example, I can't think of any either :D, i'd rather just use async-pool too!
